handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -230,42 +230,44 @@ protected void deallocate() {
         this.jdkCompatibilityMode = jdkCompatibilityMode;
         Lock readerLock = context.ctxLock.readLock();
         readerLock.lock();
+        final long ssl;
         try {
-            ssl = SSL.newSSL(context.ctx, !context.isClient());
+            this.ssl = ssl = SSL.newSSL(context.ctx, !context.isClient());
```

Comment:
@carl-mastrangelo hard to say from mobile, @jasontedor mentioned the instance is published before the constructor completes so while I cannot verify right now I would say it might be possible which is why I flagged it.